web-transport-proto/src/connect.rs

```diff
@@ -102,7 +102,12 @@ impl ConnectRequest {
         headers.set(":method", "CONNECT");
         headers.set(":scheme", self.url.scheme());
         headers.set(":authority", self.url.authority());
-        headers.set(":path", self.url.path());
+        let path_and_query = self
```

Comment:
this is the actual change, happy to use pattern matching if you prefer that

Reply:
Yeah I think the `match` is clearer but either works.

Reply:
No worries, give me a min